### DISCUSSION TITLE
Installation Help on Surface Pro 11

### DISCUSSION THREAD

**Discussion by dpodjasek** [MustardChef/WSABuilds#437] [Q&A]
I'm hoping someone can help me.  I have the Surface Pro 11 running Win11 24H2. I think I'm stuck on trying to install WSA.  I really only want it for running the full Android Kindle app, since Amazon refuses to make a proper app for the WIndows store.

Here is what I have done so far:

(1) Uninstalled the preinstalled version of WSA that came on the machine.
(2) Downloaded the "WSA_2307.40000.6.0_arm64_Release-Nightly-MindTheGapps-13.0-RemovedAmazon.7z" file.
(3) Extracted it and renamed the folder to "C:\WSA\".
(4) Run Run.bat as admin.
(5) It decompressed a bunch of things and got to the step where it said it needed to reboot to finish installing Microsoft Virtualization (pretty sure that's what it called it). I entered "y" and when it asked if I was sure I entered "a" for "yes and accept all".

The machine rebooted and said it was installing components.  When it came back I was expecting it to continue with the script.  Instead, nothing visible happened.  I left it for a bit, then decided to try running Run.bat as admin again.  This time it gives me the following:

![image](https://github.com/user-attachments/assets/b31a3fa0-8bc1-4960-8deb-de4ef024a12d)

I'm not sure what to do at this point.  It doesn't seem like anything has been added to my start menu.  Nothing shows as recently installed.  I would try uninstalling magisk, but I have no idea how to do it.

I was hoping this would be something I could get working, but it feels like there is a lot of assumed knowledge which I am definitely lacking. I'm a programmer, but I don't know much about how Android works on a technical level or what wizardry is required or this. :(

I can't even figure out where all of this got extracted to.

Can anyone give me any guidance?

Thanks!

**Comment by dpodjasek** [comment]
Well, I wiped my computer today to try this again.  This time, I made sure to enable virtual machine platform and windows hypervisor platform before installing WOA.  Rebooted then tries the WSA install again.

This time it did not force a reboot, but I end up with the same error.  It seems to be complaining about the having a previously installed copy of MagiskOnWsa.  I assume this was the preinstalled copy of WSA from MS?  I did uninstall that before trying to install, via programs in Settings.  

Does anyone have any advice for how I can completely uninstall MagiskOnWsa to get past this error?

Is there a more detailed guide anywhere that explains how to get WSA working on SP11?

Thanks!

Current screen:
![image](https://github.com/user-attachments/assets/e700e994-65d7-42e0-81bc-6f12e975b6df)

**Comment by fdkhe** [comment]
Hi, I just figured out a relatively stable way to install WSA on Surface Pro 11. I have read an [article](https://www.xda-developers.com/windows-subsystem-for-android-snapdragon-how-to/) about it and found "Sinclair was the first to discover that the actual reason WSA won't boot is because parts of it use Armv7 to run, which isn't supported by the Armv8 Snapdragon X series." There are some [patches](https://github.com/snickler/WSA-Patched/releases) for WSA in that article. 

For my own experience, I used "WSA_2407.40000.4.0_arm64_Release-Nightly-NoGApps.7z" to start. I put patch files into the WSA folder first and then clicked "run.bat". The patch v0.0.2 did not work well and patch v0.0.1 worked but had some bugs with the file system of WSA. The only way I found to fix it was that after installing WSA with patch v0.0.1, turn off WSA, replace the two files in patch v0.0.1 with same-name files in patch v0.0.2, and all set.
